scarlet-protocol-stomp/src/test/java/com/tinder/scarlet/stomp/okhttp/OkHttpStompIntegrationTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.tinder.scarlet.stomp.okhttp
+
+import com.tinder.scarlet.ProtocolEvent
+import com.tinder.scarlet.Stream
+import com.tinder.scarlet.testutils.rule.OkHttpStompWebSocketConnection
+import com.tinder.scarlet.testutils.test
+import com.tinder.scarlet.ws.Receive
+import com.tinder.scarlet.ws.Send
+import org.apache.activemq.junit.EmbeddedActiveMQBroker
+import org.junit.Rule
+import org.junit.Test
+import java.util.logging.Logger
+
+class OkHttpStompIntegrationTest {
+
+    @get:Rule
+    val broker = object : EmbeddedActiveMQBroker() {
+        override fun configure() {
+            brokerService.addConnector(BROKER_URL)
+        }
+    }
+
+    @get:Rule
+    val firstConnection = OkHttpStompWebSocketConnection.create<StompOkHttpQueueTestService>(
+        { observeProtocolEvent() },
+        OkHttpStompWebSocketConnection.Configuration(
+            login = LOGIN,
+            port = PORT,
+            password = PASSWORD,
+            host = HOST,
+            destination = DESTINATION
+        )
+    )
+
+    @get:Rule
+    val secondConnection = OkHttpStompWebSocketConnection.create<StompOkHttpQueueTestService>(
+        { observeProtocolEvent() },
+        OkHttpStompWebSocketConnection.Configuration(
+            login = LOGIN,
+            port = PORT,
+            password = PASSWORD,
+            host = HOST,
+            destination = DESTINATION
+        )
+    )
+
+    @Test
+    fun `correct receive and send messages`() {
+        val queueTextObserver = secondConnection.client.observeText().test()
+
+        firstConnection.open()
+        secondConnection.open()
+
+        firstConnection.client.sendText("message1")
+        firstConnection.client.sendText("message2")
+        firstConnection.clientClosure()
+
+
+        LOGGER.info("${queueTextObserver.values}")
+        queueTextObserver.awaitCountAtLeast(1)// because broker has a bug and it loses messages sometimes
```

Comment:
nit: https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html

Reply:
I fixed this test. This comment is extra now